src/object.rs

```diff
@@ -0,0 +1,321 @@
+use std::fmt;
+use std::collections::HashMap;
+use std::hash::{Hash,Hasher};
+use std::cell::RefCell;
+use std::rc::Rc;
+use ast;
+
+#[derive(Hash, Eq, PartialEq, Clone, Debug)]
+pub enum Object {
+    Int(i64),
+    Bool(bool),
+    String(String),
+    Return(Rc<Return>),
+    Function(Rc<Function>),
+    Builtin(Builtin),
+    Array(Rc<Array>),
+    Hash(Rc<MonkeyHash>),
+    Null,
+}
+
+impl Object {
+    pub fn inspect(&self) -> String {
```

Comment:
The book used the inspect method, but I think if I were more idiomatic I would just implement the `Display` trait.

Reply:
Display is useful for something displayable in a user-formatted way.  If you want a low level detail representation then look at the `Debug` trait, that seems closer to what `inspect` is from what I see about what `inspect` is supposed to be?

Reply:
Yeah, I think that's right. I would imagine in a more complete implementation of a language I would want to implement both (not as derived, but specific to those objects)